static/js/util/test_utils.js

```diff
@@ -21,14 +22,38 @@ export function findCourse(courseSelector: (course: Course, program: Program) =>
   throw "Unable to find course";
 }
 
-export const alterFirstRun = (course: Course, overrideObject: Object): Course => {
+export const alterFirstRun = (course: Course, overrideObject: Object): CourseRun => {
   course.runs[0] = Object.assign({}, course.runs[0], overrideObject);
+  return course.runs[0];
 };
 
 export function findAndCloneCourse(courseSelector: (course: Course, program: Program) => boolean): Course {
   return _.cloneDeep(findCourse(courseSelector));
 }
 
+export function setCourseRunCount(course: Course, desiredRunCount: number, courseRunToCopy: ?CourseRun) {
+  let currentRunCount = course.runs.length;
+  if (currentRunCount < desiredRunCount) {
+    let courseRun = currentRunCount === 0 ? courseRunToCopy : course.runs[0];
+    if (!courseRun) {
+      throw new Error('Need a course run to copy.');
+    }
+    let runsNeeded = desiredRunCount - currentRunCount;
+    let idMax = _.max(_.map(course.runs, run => run.id)) || 0;
```

Comment:
this is creating an entirely new course object. it's not changing/altering the existing values in that giant`DASHBOARD_RESPONSE` variable, nor does it need to be concerned about the other values in there. i wrote this for the convenience of writing these tests, but it's also an intermediate step toward handling test data like we should, ie: generating the test data we need on the fly instead of depending on a giant static object that satisfies every possible condition that we want to test for. the need to check for unique id's/etc in that variable is just one symptom of the anti-pattern